service/frontend/service.go

```diff
@@ -405,14 +407,19 @@ func namespaceRPS(
 	frontendResolver membership.ServiceResolver,
 	namespace string,
 ) float64 {
-	globalRPS := float64(globalRPSFn(namespace))
+	globalRPS := globalRPSFn(namespace)
+	hostRPS := perInstanceRPSFn(namespace)
+
+	return effectiveRPS(frontendResolver, hostRPS, globalRPS)
+}
+
+func effectiveRPS(frontendResolver membership.ServiceResolver, hostRPS int, globalRPS int) float64 {
```

Comment:
nit: looks like hostRPS fn doesn't needs to be evaluated when globalRPS and resolver are available.

Reply:
Made it lazy